src/eve/datamodels.py

```diff
@@ -81,17 +86,29 @@
     TypeVar,
     Union,
 )
-
-import attr
-
-from eve import typingx, utils
-from eve.concepts import NOTHING
-from eve.typingx import NonDataDescriptor
+from eve.type_definitions import NOTHING
 
 
 # Typing
 T = TypeVar("T")
 V = TypeVar("V")
+T_contra = TypeVar("T_contra", contravariant=True)
+V_co = TypeVar("V_co", covariant=True)
+
+
+class NonDataDescriptor(Protocol[T_contra, V_co]):
+    @typing.overload
+    def __get__(self, _instance: None, _owner_type: Type[T_contra]) -> NonDataDescriptor:
+        ...
+
+    @typing.overload
+    def __get__(self, _instance: T_contra, _owner_type: Optional[Type[T_contra]] = None) -> V_co:
+        ...
+
+
+class DataDescriptor(NonDataDescriptor[T_contra, V], Protocol):
```

Comment:
These definitions have been moved here from the deleted typingx module just to avoid further changes in this PR unrelated to the typing infrastructure and they will be likely removed in the next PR.